src/effects/builtin/graphiceq/graphiceqbandsmodel.cpp

```diff
@@ -0,0 +1,160 @@
+/*
+ * Audacity: A Digital Audio Editor
+ */
+#include "graphiceqbandsmodel.h"
+
+#include "graphiceq.h"
+
+#include "log.h"
+
+namespace au::effects {
+namespace {
+QString centerFrequencyToString(double frequency)
+{
+    if (frequency < 1000.0) {
+        return QString::number(static_cast<int>(frequency + .5));
+    } else {
+        const auto khz = frequency / 1000.0;
+        return QString::number(khz, 'f', (khz == static_cast<int>(khz)) ? 0 : 1) + "k";
+    }
+}
+}
+
+GraphicEqBandsModel::GraphicEqBandsModel(QObject* parent, std::function<GraphicEq* ()> eqGetter)
+    : QAbstractListModel(parent),  m_getEq(std::move(eqGetter)), mSliders([this]() -> EqualizationCurvesList* {
+    GraphicEq* const eq = m_getEq();
```

Comment:
Is the reason for such complications (passing a lambda to get a lambda to fetch a value) that the model is created before the effect instance is accessible?

Isn't it possible to delay the initialization, and use the original approach with references?